src/App.tsx

```diff
@@ -1,24 +1,153 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos, addTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useTodoContext } from './context';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 12113;
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    setAllTodos,
+    visibleTodos,
+    errorHandler,
+    errorMessage,
+    setErrorMessage,
+    inputRef,
+  } = useTodoContext();
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    const trimmedTitle = query.trim();
+
+    if (trimmedTitle === '') {
+      errorHandler('Title should not be empty');
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      setTempTodo({
+        id: 0,
+        userId: USER_ID,
+        title: trimmedTitle,
+        completed: false,
+      });
+
+      const addedTodo = await addTodo({
+        userId: USER_ID,
+        title: trimmedTitle,
+        completed: false,
+      });
+
+      setAllTodos((prevTodos: Todo[] | null) => {
+        return prevTodos ? [...prevTodos, addedTodo] : [addedTodo];
+      });
+
+      setTempTodo(null);
+      setQuery('');
+      setIsLoading(false);
+    } catch (error) {
+      setTempTodo(null);
+      setIsLoading(false);
+      errorHandler('Unable to add Todo');
+    }
+  };
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const todos = await getTodos(USER_ID);
+
+        setAllTodos(todos);
+      } catch (error) {
+        errorHandler('Unable to load todos');
+      }
+    };
+
+    loadTodos();
+  }, [setAllTodos]);
```

Comment:
Lint error: React Hook useEffect has a missing dependency: 'errorHandler'. Either include it or remove the dependency array